src/execution_plan/ops/op_expand_into.c

```diff
@@ -156,10 +157,16 @@ static Record _handoff(OpExpandInto *op) {
 		// Current record resides at row recordCount.
 		int rowIdx = op->recordCount;
 		op->r = op->records[op->recordCount];
-		assert(Record_GetType(op->r, op->srcNodeIdx) == REC_TYPE_NODE);
-		assert(Record_GetType(op->r, op->destNodeIdx) == REC_TYPE_NODE);
 		srcNode = Record_GetNode(op->r, op->srcNodeIdx);
 		destNode = Record_GetNode(op->r, op->destNodeIdx);
+		if(!srcNode || !destNode) {
+			// An endpoint may not resolve if an Optional op failed to match,
+			// in which case we can skip this record.
+			// Mark as NULL to avoid double free.
+			op->records[op->recordCount] = NULL;
+			continue;
+
+		}
```

Comment:
1. `srcNode` is guarantee to exists, see `consume` function
2. it is wasteful to see if `destNode` exists in the record at this point in time, it would be much better to check for `destNode` within the consume function. we would like to consider records where both src and dest exists.